### PR TITLE
Feature/add redis caching and style tidyup

### PR DESCRIPTION
**Pull request by jimcroft**

### CODE REVIEW

File: lib/hiera/backend/cloudformation_backend.rb [hunk 1]
```diff
@@ -1,187 +1,227 @@
...
+        @cache_ttl = cache_ttl
+
+        if Config.include?(:cloudformation) && !Config[:cloudformation].nil?
+          if Config[:cloudformation].include?(:redis_hostname)
+            @redis_hostname = Config[:cloudformation][:redis_hostname]
+          end
+        end
+
+        if @redis_hostname
+          @redis = redis.StrictRedis(@redis_hostname, 6379, 0)
```
Comment (sigmaris) [inline lib/hiera/backend/cloudformation_backend.rb]: image

Comment (jimcroft) [inline lib/hiera/backend/cloudformation_backend.rb]: Context switching is fun in meatspace!

File: lib/hiera/backend/cloudformation_backend.rb [hunk 2]
```diff
@@ -1,187 +1,261 @@
...
-						return json_object.to_i
-					end
-				else
-					return json_object
-				end
-			end
-		end
-	end
+        if @redis_hostname
+          @redis = Redis.new(:host => @redis_hostname, :port => 6379, :db => 0)
```
Comment (sigmaris) [inline lib/hiera/backend/cloudformation_backend.rb]: The port and database number should also be configurable (running multiple redis servers on different ports is fairly common)

**Comment by sigmaris** [comment]
Apart from one comment, looks good. Can you also update the copyright notices in any modified files to 'Copyright 2013-2015 FanDuel Ltd'.

**Comment by sigmaris** [comment]
cool :+1:

**Comment by sigmaris** [comment]
:+1:

**Pull request merged**